client/src/utils/to-learn-path.ts

```diff
@@ -1,18 +1,33 @@
 import { withPrefix } from 'gatsby';
 
 interface ToLearnPathKwargs {
-  superBlock: string;
-  block: string;
-  challenge: string;
+  block?: string;
+  challenge?: string;
+  superBlock?: string;
 }
+
+/**
+ * Builds a learning url path from folders.
+ * - /learn/:superBlock/:block/:challenge
+ * @params {Object} folders.
+ * @params {string} [folders.superBlock] a folder.
+ * @params {string} [folders.block] a folder.
+ * @params {string} [folders.challenge] a folder.
+ * @returns {string} A learning url path.
+ */
```

Comment:
The function takes a single param instead of four, so I think the param list here is not accurate. Furthermore, the function signature is pretty clear already, so I'm not sure if we really need to add JSDoc to it. Though I do appreciate the description 🙂 

This is me hovering on the function from `challenges.tsx`

<img width="812" alt="Screen Shot 2021-07-30 at 13 14 41" src="https://user-images.githubusercontent.com/25715018/127609148-646da86a-89ac-4126-b062-67c2f484d6e0.png">

```suggestion
/**
 * Builds a learning url path from folders.
 * - /learn/:superBlock/:block/:challenge
 */
```

Reply:
The doc section has been update:
- Documentation contains one `@param` statement
- Kept the `@return` statement

Note: the other `@param` statements were the contents of the Object not individual params.

@huyenltnguyen, please let me know if this works. :+1: